### PR TITLE
coprocessor/scanner: set key_only=true if needed

### PR DESCRIPTION
**Pull request by AndreMouche**
@BusyJay @hicqu @hanfei1991  PTAL

### CODE REVIEW

**Comment by hicqu** [comment]
LGTM.

**Comment by zhangjinpeng1987** [comment]
ci failed.

File: src/coprocessor/dag/executor/table_scan.rs [hunk 1]
```diff
@@ -91,7 +91,13 @@ impl TableScanExecutor {
     }
 
     fn new_scanner(&self, range: KeyRange) -> Result<Scanner> {
-        Scanner::new(&self.store, ScanOn::Table, self.desc, false, range).map_err(Error::from)
+        Scanner::new(
+            &self.store,
+            ScanOn::Table,
+            self.desc,
+            self.col_ids.is_empty(),
```
Comment (BusyJay) [inline src/coprocessor/dag/executor/table_scan.rs]: What about handle?

Comment (AndreMouche) [reply src/coprocessor/dag/executor/table_scan.rs]: `handle` wouldn't appear in `self.col_ids` since it was filtered on L47 @BusyJay

**Comment by hanfei1991** [comment]
LGTM

**Comment by AndreMouche** [comment]
Performance test result
tidb branch(https://github.com/pingcap/tidb/pull/5447):

in master
```
MySQL [test]> select count(*) from table10;
+-----------+
| count(*)  |
+-----------+
| 264488600 |
+-----------+
1 row in set (1 min 31.44 sec)

MySQL [test]> select count(*) from table10;
+-----------+
| count(*)  |
+-----------+
| 264488600 |
+-----------+
1 row in set (1 min 31.54 sec)

MySQL [test]> select count(*) from table10;
+-----------+
| count(*)  |
+-----------+
| 264488600 |
+-----------+
1 row in set (1 min 31.30 sec)

MySQL [test]> select count(*) from table10;
+-----------+
| count(*)  |
+-----------+
| 264488600 |
+-----------+
1 row in set (1 min 31.19 sec)
```

current branch

```
MySQL [test]> select count(*) from table10;
+-----------+
| count(*)  |
+-----------+
| 264488600 |
+-----------+
1 row in set (1 min 1.98 sec)

MySQL [test]> select count(*) from table10;
+-----------+
| count(*)  |
+-----------+
| 264488600 |
+-----------+
1 row in set (1 min 2.04 sec)

MySQL [test]> select count(*) from table10;
+-----------+
| count(*)  |
+-----------+
| 264488600 |
+-----------+
1 row in set (1 min 2.04 sec)

MySQL [test]> select count(*) from table10;
+-----------+
| count(*)  |
+-----------+
| 264488600 |
+-----------+
1 row in set (1 min 1.92 sec)
```

**Pull request merged**